atomic-swap/src/main/java/nl/tudelft/trustchain/atomicswap/swap/eth/EthereumSwap.kt

```diff
@@ -0,0 +1,129 @@
+package nl.tudelft.trustchain.atomicswap.swap.eth
+
+import android.annotation.SuppressLint
+import android.util.Log
+import io.reactivex.plugins.RxJavaPlugins
+import nl.tudelft.ipv8.util.sha256
+import nl.tudelft.ipv8.util.toHex
+import nl.tudelft.trustchain.atomicswap.swap.Trade
+import nl.tudelft.trustchain.atomicswap.ui.swap.LOG
+import org.web3j.crypto.Credentials
+import org.web3j.protocol.Web3j
+import org.web3j.protocol.core.DefaultBlockParameter
+import org.web3j.protocol.core.DefaultBlockParameterName
+import org.web3j.protocol.core.DefaultBlockParameterNumber
+import org.web3j.protocol.core.methods.response.TransactionReceipt
+import org.web3j.tx.RawTransactionManager
+import org.web3j.tx.gas.DefaultGasProvider
+import org.web3j.utils.Convert
+import java.math.BigInteger
+
+
+@SuppressLint("CheckResult")
```

Comment:
Why is this suppressed? Add a comment explaining it if required.

Reply:
I removed the suppresion and fixed the warning. The problem was that a resource wasn't being disposed.